iroh-gateway/src/config.rs

```diff
@@ -24,21 +23,19 @@ impl Config {
             writeable,
             fetch,
             cache,
-            headers: HashMap::new(),
+            headers: HeaderMap::new(),
             port,
         }
     }
 
     pub fn set_default_headers(&mut self) {
-        let mut headers = HashMap::new();
-        headers.insert(ACCESS_CONTROL_ALLOW_ORIGIN.to_string(), "*".to_string());
-        headers.insert(ACCESS_CONTROL_ALLOW_HEADERS.to_string(), "*".to_string());
-        headers.insert(ACCESS_CONTROL_ALLOW_METHODS.to_string(), "*".to_string());
-        headers.insert(
-            CACHE_CONTROL.to_string(),
-            "no-cache, no-transform".to_string(),
-        );
-        headers.insert(ACCEPT_RANGES.to_string(), "none".to_string());
+        let mut headers = HeaderMap::new();
+        headers.insert(ACCESS_CONTROL_ALLOW_ORIGIN, VALUE_STAR.clone());
+        headers.insert(ACCESS_CONTROL_ALLOW_HEADERS, VALUE_STAR.clone());
+        headers.insert(ACCESS_CONTROL_ALLOW_METHODS, VALUE_STAR.clone());
+        // todo(arqu): remove these once propperly implmented
+        headers.insert(CACHE_CONTROL, VALUE_NO_CACHE_NO_TRANSFORM.clone());
+        headers.insert(ACCEPT_RANGES, VALUE_NONE.clone());
```

Comment:
Maybe using https://docs.rs/headers/0.3.7/headers/trait.HeaderMapExt.html would be simpler. You could do:
```rust
  headers.insert_typed(AccessControlAllowOrigin::ANY);
  ...
```
and not have to declare a bunch of statics yourself.

Reply:
Eh... tried it, but it's not great
```
        headers.typed_insert(AccessControlAllowOrigin::ANY);
        headers.typed_insert(AccessControlAllowMethods::from_iter(
            vec![
                Method::GET,
                Method::PUT,
                Method::POST,
                Method::DELETE,
                Method::HEAD,
                Method::OPTIONS,
            ]
            .into_iter(),
        ));
        headers.typed_insert(AccessControlAllowHeaders::...);
```

For most types I have to manually add each one and then later on maintain it, ANY is not an option for a lot of those.
I'll make a combination for where it makes sense and where it doesn't.

Reply:
It get's worse once I start mixing stuff. How about I keep the constants &str and then make `HeaderName` every time I use em?

Reply:
Well, to each its own. I actually like the explicit enumeration. Also I think you don't need the `.into_iter()` after the `vec![]`. Your call anyway.

Reply:
this is a little nicer and should work fine 

```rust
headers.typed_insert(AccessControlAllowOrigin::ANY);
headers.typed_insert([
    Method::GET,
    Method::PUT,
    Method::POST,
    Method::DELETE,
    Method::HEAD,
    Method::OPTIONS,
].into_iter().collect::<AccessControlAllowMethods>()));
````

but I agree, making these explicit is good, especially as we will want to make these methos configurable later on anyway